packages/dbml-core/src/model_structure/endpoint.js

```diff
@@ -22,28 +23,35 @@ class Endpoint extends Element {
       this.error(`Can't find table ${shouldPrintSchema(schema)
         ? `"${schema.name}".` : ''}"${tableName}"`);
     }
-    const field = table.findField(fieldName);
-    this.setField(field, table);
+    this.setFields(fieldName, table) 
   }
 
   generateId () {
     this.id = this.dbState.generateId('endpointId');
   }
 
   equals (endpoint) {
-    return this.field.id === endpoint.field.id;
+    if (this.fields.length != endpoint.fields.length) return false;
+    return this.fields.length == 1 ? this.field.id === endpoint.field.id : this.compareFields(endpoint);
```

Comment:
same here, `===` instead of `==`

Reply:
Why we have to divide into 2 cases for comparing fields here? Can `compareFields` compare fields with length 1?